### PR TITLE
Easy to switch default HTTP protocol

### PR DESCRIPTION
**Pull request by anisimkov**
TN: S507-051

### CODE REVIEW

**Comment by TurboGit** [comment]
Not sure about this. It means that every calls using HTTP/1 today will automatically be changed to HTTP/2 when we will be changing the default. For server part it makes sense as it supports both protocols. But for the client side we either do an HTTP/1 or and HTTP/2 call. If the server is not ready (supporting) HTTP/2 the same program will fail !

**Comment by anisimkov** [comment]
OK. I made separate HTTP_Default for client part, not depend on server part.

File: src/core/aws-client.ads [hunk 1]
```diff
@@ -271,24 +273,25 @@ package AWS.Client is
...
+      Pwd          : String          := No_Data;
+      Proxy        : String          := No_Data;
+      Proxy_User   : String          := No_Data;
+      Proxy_Pwd    : String          := No_Data;
+      Timeouts     : Timeouts_Values := No_Timeout;
+      Headers      : Header_List     := Empty_Header_List;
+      Progress     : access procedure
+                       (Total, Sent : Stream_Element_Offset) := null;
+      User_Agent   : String          := Default.User_Agent;
+      HTTP_Version : HTTP_Protocol   := HTTP_Default)
```
Comment (TurboGit) [inline src/core/aws-client.ads:294]: Please remove this option it is just not supported. And I would add a Pre condition on Upload with connection to check that it is HTTP/1 for now.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Minor change !

File: src/core/aws-client.ads (hunk 1 above)
Comment (anisimkov) [reply src/core/aws-client.ads]: Done

**Pull request merged**